modules/tfm/tfm/boards/common/attest_hal.c

```diff
@@ -15,34 +15,205 @@
-{
+/* 16 is the only supported length */
+#define CONFIG_SB_PUBLIC_KEY_HASH_LEN 16
+
+/* implementation_id is a mandatory IAT claim */
+#define CONFIG_SB_IMPLEMENTATION_ID 1
+#include "bl_storage.h"
+#include <nrfx_nvmc.h>
+
+
+static enum tfm_security_lifecycle_t map_bl_storage_lcs_to_tfm_slc(enum lcs lcs){
 	switch (lcs) {
-	case PLAT_OTP_LCS_ASSEMBLY_AND_TEST:
+	case ASSEMBLY:
 		return TFM_SLC_ASSEMBLY_AND_TEST;
-	case PLAT_OTP_LCS_PSA_ROT_PROVISIONING:
+	case PROVISION:
 		return TFM_SLC_PSA_ROT_PROVISIONING;
-	case PLAT_OTP_LCS_SECURED:
+	case SECURE:
 		return TFM_SLC_SECURED;
-	case PLAT_OTP_LCS_DECOMMISSIONED:
+	case DECOMMISSIONED:
 		return TFM_SLC_DECOMMISSIONED;
-	case PLAT_OTP_LCS_UNKNOWN:
+	case UNKNOWN:
 	default:
 		return TFM_SLC_UNKNOWN;
 	}
 }
 
+static enum lcs map_tfm_slc_to_bl_storage_lcs(enum tfm_security_lifecycle_t lcs){
+	switch (lcs) {
+	case TFM_SLC_ASSEMBLY_AND_TEST:
+		return ASSEMBLY;
+	case TFM_SLC_PSA_ROT_PROVISIONING:
+		return PROVISION;
+	case TFM_SLC_SECURED:
+		return SECURE;
+	case TFM_SLC_DECOMMISSIONED:
+		return DECOMMISSIONED;
+	case TFM_SLC_UNKNOWN:
+	default:
+		return UNKNOWN;
+	}
+}
+
+/* This is temporary solution until the bl_storage API is available in TF-M
+ * This points to the UICR OTP region
+ */
+static const struct bl_storage_data *p_bl_storage_data =
+	(struct bl_storage_data *)PROVISION_PARTITION_ADDRESS;
+
+#define STATE_ENTERED 0x0000
+#define STATE_NOT_ENTERED 0xFFFF
+#define	EINVAL 22	/* Invalid argument */
+
+
+/** Function for reading a half-word (2 byte value) from OTP.
+ *
+ * \details The flash in OTP supports only aligned 4 byte read operations.
+ *          This function reads the encompassing 4 byte word, and returns the
+ *          requested half-word.
+ *
+ * \param[in]  ptr  Address to read.
+ *
+ * \return value
+ */
+static uint16_t read_halfword(const uint16_t *ptr)
+{
+	bool top_half = ((uint32_t)ptr % 4); /* Addr not div by 4 */
+	uintptr_t target_addr = (uintptr_t)ptr & ~3; /* Floor address */
+	uint32_t val32 = *(uint32_t *)target_addr;
+	__DSB(); /* Because of nRF9160 Erratum 7 */
+
+	return (top_half ? (val32 >> 16) : val32) & 0x0000FFFF;
+}
+
+/** Function for writing a half-word (2 byte value) to OTP.
+ *
+ * \details The flash in OTP supports only aligned 4 byte write operations.
+ *          This function writes to the encompassing 4 byte word, masking the
+ *          other half-word with 0xFFFF so it is left untouched.
+ *
+ * \param[in]  ptr  Address to write to.
+ * \param[in]  val  Value to write into @p ptr.
+ */
+static void write_halfword(const uint16_t *ptr, uint16_t val)
+{
+	bool top_half = (uint32_t)ptr % 4; /* Addr not div by 4 */
+	uint32_t target_addr = (uint32_t)ptr & ~3; /* Floor address */
+
+	uint32_t val32 = (uint32_t)val | 0xFFFF0000;
+	uint32_t val32_shifted = ((uint32_t)val << 16) | 0x0000FFFF;
+
+	nrfx_nvmc_word_write(target_addr, top_half ? val32_shifted : val32);
+}
```

Comment:
Let's send a PR to nrfx to introduce nrfx_nvmc_otp_halfword_write.

We can use write_halfword in the mean time.

Reply:
Actually, they already have a function for this.

Reply:
Fixed in commit that will come soon.